CleanWeather/Scenes/FavouriteCities/FavouriteCitiesWorker.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  FavouriteCitiesWorker.swift
+//  CleanWeather
+//
+//  Created by Marek Skrzelowski on 05/12/2019.
+//  Copyright © 2019 vandermesis. All rights reserved.
+//
+
+import Foundation
+
+protocol FavouriteCitiesWorker {
+    func fetchCity(completion: FetchCitiesCompletion?)
+    func saveFavouriteCity(city: FavouriteCity)
+    func loadFavouriteCities() -> [FavouriteCity]
+}
+
+final class FavouriteCitiesWorkerImpl {
+
+    private let networking: WeatherNetworking
```

Comment:
you could make all the methods in worker asynchronous, so do not return data, but add a completion just like for fetching

so in future we can simply add networking and it can work

Reply:
nie poprawiłeś tego

Reply:
I did change fetchFavouriteCities so it have completion and it works. But what I've noticed that if I really simulate asynchronous situation - for example set completion of fetching from database to wait 2 sec.  and now netoworking has 1 sec. - I have problem in FavouriteCitiesInteractor becasue getCities and didSelectCity methods depends of both networking and database. I managed make getCities() working properly with DispatchGroup() but couldn't manage didSelectCity to work :( I didn't commit those DispatchGroup() trial and errors, cause I think I got that all wrong. Should we use some kind of syncing of both asynchronous threads here?